sql/core/src/test/scala/org/apache/spark/sql/execution/command/DDLSuite.scala

```diff
@@ -1405,6 +1405,31 @@ abstract class DDLSuite extends QueryTest with DDLSuiteBase {
     }
   }
 
+  test("SPARK-51747: When legacy conf enabled data source read ignores option change") {
+    withSQLConf(SQLConf.READ_DATA_SOURCE_CACHE_IGNORE_OPTIONS.key -> "true") {
+      withTable("t") {
+        spark.sql("CREATE TABLE t(a string, b string) USING CSV".stripMargin)
+        spark.sql("INSERT INTO TABLE t VALUES ('a;b', 'c')")
+        spark.sql("INSERT INTO TABLE t VALUES ('hello; world', 'test')")
+
+        // check initial contents of table
+        checkAnswer(spark.table("t"), Row("a;b", "c") :: Row("hello; world", "test") :: Nil)
```

Comment:
It would be great to check the cached plan after this first read as well. We should ensure plan are cached